packages/custom-token-bridging/scripts/exec-viaLib.js

```diff
@@ -0,0 +1,108 @@
+
+const l1Provider = new providers.JsonRpcProvider(process.env.L1RPC)
+const l2Provider = new providers.JsonRpcProvider(process.env.L2RPC)
+ 
+const l1Wallet = new Wallet(walletPrivateKey, l1Provider)
+const l2Wallet = new Wallet(walletPrivateKey, l2Provider)
+
+/**
+* Set the initial supply of L1 token that we want to bridge
+* Note that you can change the value 
+*/
+const premine = ethers.utils.parseEther("3")
+
+const main = async () => {
+  await arbLog('Setting Up Your Token With The Generic Custom Gateway Using arb-ts Library')
+
+  /**
+   * Use l2Network to create an arb-ts AdminErc20Bridger instance
+   * We'll use AdminErc20Bridger for its convenience methods around registering tokens to the custom gateway
+   */
+  const l2Network = await getL2Network(l2Provider)
+  const adminTokenBridge = new AdminErc20Bridger(l2Network)
+
+  const l1Gateway = l2Network.tokenBridge.l1CustomGateway
+  const l1Router = l2Network.tokenBridge.l1GatewayRouter
+  const l2Gateway= l2Network.tokenBridge.l2CustomGateway
+
+  /**
+  * Deploy our custom token smart contract to L1
+  * We give the custom token contract the address of l1CustomGateway and l1GatewayRouter as well as the initial supply (premine) 
+  */
+  const L1CustomToken = await ( await ethers.getContractFactory('L1Token') ).connect(l1Wallet)
+  console.log('Deploying cutsom token to L1')
+  const l1CustomToken = await L1CustomToken.deploy(l1Gateway, l1Router, premine)
+  await l1CustomToken.deployed()
+  console.log(`custom token is deployed to L1 at ${l1CustomToken.address}`)
+
+  /**
+  * Deploy our custom token smart contract to L2
+  * We give the custom token contract the address of l2CustomGateway and our l1CustomToken 
+  */
+  const L2CustomToken = await ( await ethers.getContractFactory('L2Token') ).connect(l2Wallet)
+  console.log('Deploying cutsom token to L2')
+  const l2CustomToken = await L2CustomToken.deploy(l2Gateway,l1CustomToken.address)
+  await l2CustomToken.deployed()
+  console.log(`custom token is deployed to L2 at ${l2CustomToken.address}`)
+
+  console.log("Registering custom token on L2:")
+
+  /**
+  * ٍRegister custom token on our custom gateway
+  */
+  const registerTokenTx = await adminTokenBridge.registerCustomToken(
+    l1CustomToken.address,
+    l2CustomToken.address,
+    l1Wallet,
+    l2Provider
+  )
+  
+  const registerTokenRec = await registerTokenTx.wait()
+  console.log(
+    `Registering token txn confirmed on L1! 🙌 L1 receipt is: ${registerTokenRec.transactionHash}`
+  )
+
+  /**
+   * The L1 side is confirmed; now we listen and wait for the for the Sequencer to include the L2 side; we can do this by computing the expected txn hash of the L2 transaction.
+   * To compute this txn hash, we need our message's "sequence numbers", unique identifiers of each L1 to L2 message. We'll fetch them from the event logs with a helper method
+   */
+  const l1ToL2Msgs = await registerTokenRec.getL1ToL2Messages(l2Provider)
```

Comment:
we can instead use `registerTokenTx.waitForL2()` which should be available from [here](https://github.com/OffchainLabs/arbitrum/blob/7bab2e55db008c2ba34d201eab514cb46b10c708/packages/arb-ts/src/lib/message/L1Transaction.ts#L289).
cc @yahgwai

Reply:
here is an example flow https://github.com/OffchainLabs/arbitrum/blob/6ae9b968f19ff5e581fd98e935292d8442bccb15/packages/arb-ts/integration_test/customerc20.test.ts#L263-L265

Reply:
this one we actually cant, since the result has multiple messages on it we want to check each one